### ISSUE TITLE
ACPI call trace in suspend-resume

### ISSUE THREAD

**Issue by plbossart** [thesofproject/linux#3627]
Between 5.18-rc1 and 5.18-rc2, the following call trace started appearing on WHL UpExtreme

[dmesg.log](https://github.com/thesofproject/linux/files/8606234/dmesg.log)

**Comment by plbossart** [comment]
````
[   47.620086] ACPI: PM: Preparing to enter system sleep state S3
[   47.621392] ACPI: PM: Saving platform NVS memory
[   47.621415] Disabling non-boot CPUs ...
[   47.623767] smpboot: CPU 1 is now offline
[   47.627608] unchecked MSR access error: WRMSR to 0x123 (tried to write 0x0000000000000001) at rIP: 0xffffffff979c7eaf (restore_processor_state+0x24f/0x350)
[   47.627621] Call Trace:
[   47.627624]  <TASK>
[   47.627633]  x86_acpi_suspend_lowlevel+0x10c/0x180
[   47.627641]  acpi_suspend_enter+0x5f/0x260
[   47.627647]  ? _raw_spin_unlock_irqrestore+0x33/0x50
[   47.627654]  suspend_devices_and_enter+0x7d2/0x910
[   47.627665]  pm_suspend.cold+0x352/0x39d
[   47.627672]  state_store+0x71/0xd0
[   47.627679]  kernfs_fop_write_iter+0x13c/0x1f0
[   47.627687]  new_sync_write+0x109/0x190
[   47.627700]  vfs_write+0x287/0x310
[   47.627708]  ksys_write+0x68/0xe0
[   47.627715]  do_syscall_64+0x38/0x90
[   47.627721]  entry_SYSCALL_64_after_hwframe+0x44/0xae
[   47.627725] RIP: 0033:0x7f0cd4e1dc27
[   47.627730] Code: 0d 00 f7 d8 64 89 02 48 c7 c0 ff ff ff ff eb b7 0f 1f 00 f3 0f 1e fa 64 8b 04 25 18 00 00 00 85 c0 75 10 b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 51 c3 48 83 ec 28 48 89 54 24 18 48 89 74 24
[   47.627733] RSP: 002b:00007ffd51751aa8 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
[   47.627738] RAX: ffffffffffffffda RBX: 0000000000000004 RCX: 00007f0cd4e1dc27
[   47.627741] RDX: 0000000000000004 RSI: 0000557836c006b0 RDI: 0000000000000004
[   47.627743] RBP: 0000557836c006b0 R08: 0000000000000000 R09: 0000557836c006b0
[   47.627745] R10: 0000000000000077 R11: 0000000000000246 R12: 0000000000000004
[   47.627747] R13: 0000557836bfe510 R14: 00007f0cd4ef84a0 R15: 00007f0cd4ef78a0
[   47.627761]  </TASK>
[   47.627765] ACPI: PM: Low-level resume complete
[   47.627813] ACPI: PM: Restoring platform NVS memory
[   47.630862] Enabling non-boot CPUs ...
[   47.630951] x86: Booting SMP configuration:
[   47.630952] smpboot: Booting Node 0 Processor 1 APIC 0x2
[   47.634710] CPU1 is up
[   47.635161] ACPI: PM: Waking up from system sleep state S3
````

**Comment by plbossart** [comment]
````
e2a1256b17b16f9b9adf1b6fea56819e7b68e463 is the first bad commit
commit e2a1256b17b16f9b9adf1b6fea56819e7b68e463
Author: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>
Date:   Mon Apr 4 17:35:45 2022 -0700

    x86/speculation: Restore speculation related MSRs during S3 resume
    
    After resuming from suspend-to-RAM, the MSRs that control CPU's
    speculative execution behavior are not being restored on the boot CPU.
    
    These MSRs are used to mitigate speculative execution vulnerabilities.
    Not restoring them correctly may leave the CPU vulnerable.  Secondary
    CPU's MSRs are correctly being restored at S3 resume by
    identify_secondary_cpu().
    
    During S3 resume, restore these MSRs for boot CPU when restoring its
    processor state.
    
    Fixes: 772439717dbf ("x86/bugs/intel: Set proper CPU features and setup RDS")
    Reported-by: Neelima Krishnan <neelima.krishnan@intel.com>
    Signed-off-by: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>
    Tested-by: Neelima Krishnan <neelima.krishnan@intel.com>
    Acked-by: Borislav Petkov <bp@suse.de>
    Reviewed-by: Dave Hansen <dave.hansen@linux.intel.com>
    Cc: stable@vger.kernel.org
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

 arch/x86/power/cpu.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
````

**Comment by plbossart** [comment]
Fixed by PR #3629
